database/postgresql_adapter.go

```diff
@@ -52,3 +62,133 @@ func (a *PostgreSqlAdapter) DoMigrations(db *sqlx.DB, cfg *config.DbConfig) erro
 
 	return nil
 }
+
+func (a *PostgreSqlAdapter) GetDBx() *sqlx.DB {
+	if a.db == nil {
+		panic("connection to the database has not been established")
+	}
+	return a.db
+}
+
+func (a *PostgreSqlAdapter) ImportHeaders(inputFile *os.File, log *zerolog.Logger) (int, error) {
+	// prepare db for bulk insterts
+	restoreIndexes, err := a.dropTableIndexes(sql.HeadersTableName)
+	if err != nil {
+		return 0, err
+	}
+	defer func() {
+		if err = restoreIndexes(); err != nil {
+			log.Error().Msg(err.Error())
+			os.Exit(1)
+		}
+	}()
+
+	if _, err := inputFile.Seek(0, 0); err != nil {
+		return 0, err
+	}
+
+	reader := csv.NewReader(inputFile)
+	_, err = reader.Read() // Skipping the column headers line
+	if err != nil {
+		log.Error().Msg(err.Error())
+		os.Exit(1)
```

Comment:
Since the function `ImportHeaders` can return an error and because it is used in `main.go` inside:
```golang
db, err := database.Init(cfg, log)
if err != nil {
    log.Error().Msgf("cannot setup database because of error: %v", err)
    os.Exit(1)
}
```
...where the error logging and `os.Exit` is performed. I'm thinking whether just returning an error here would be more consistent.

Reply:
changed